### ISSUE TITLE
Some messages are delivered multiple times.

### ISSUE THREAD

**Issue by remogloor** [ParticularLabs/NServiceBus.ActiveMQ#9]
In failover scenarios messages are processed sucessfully multiple times. This is not an NSB bug but caused by a bug in the ActiveMQ NMS client. V1.6 has some fixes for that problem. But unfortunatly they didn't fix it properly so that there is additional work then just porting alle the patches we did for 1.5.6 to 1.6

We are currently working on this and hope to have an NMS client that fixes this issue soon.

**Comment by johnsimons** [comment]
> We are currently working on this and hope to have an NMS client that fixes this issue soon

Has this bug been reported ? Link?

Please lets **not** add more patches to our version of NMS client!

**Comment by remogloor** [comment]
https://issues.apache.org/jira/browse/AMQNET-431

We had this issue in production. They improved that situation in 1.6 so we wanted to switch to that version. But unfortunatly it has only 5/9 patches we contributed applied. We are currently proting the remaining 4 patches to 1.6 to recontribute a patch. But that is not easy as the code has major changes because of the failover implementation. But while porting the patches to 1.6 we had to analyze and understand the new failover code. While doing this we revealed several problems with the failover implementation and are working on unit tests and patches for these problems so that we can contribute them to the NMS client project.
